src/implementations/twig/components/button/button.story.js

```diff
@@ -11,117 +9,162 @@ import dataCall from '@ecl/specs-component-button/demo/data--call';
+    data.icon.name = args.icon_name;
+    data.icon.type = 'ui';
+    data.icon.size = 'xs';
+    data.icon.transform = args.icon_transform;
+    data.icon.path = defaultSprite;
+    data.icon_position = args.icon_position;
+  } else if (args.icon_name == null && data.icon) {
+    delete data.icon;
+  }
 
   return data;
 };
 
 export default {
   title: 'Components/Button',
-  decorators: [withKnobs, withCode, withNotes],
+  argTypes: {
+    label: {
+      name: 'label',
+      type: { name: 'string', required: true },
+      defaultValue: '',
+      description: 'The main label of the button',
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: '' },
+        category: 'Content',
+      },
+      control: {
+        type: 'text',
+      },
+    },
+    icon_name: {
+      name: 'icon name',
+      type: { name: 'select', required: false },
+      defaultValue: '',
+      description: 'Button icon',
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: '' },
+        category: 'Icon',
+      },
+      control: {
+        type: 'select',
+        options: iconsList,
+      },
+    },
+    icon_transform: {
+      name: 'icon transform',
+      type: { name: 'select', required: false },
+      defaultValue: '',
+      description: 'Button icon transform',
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: '' },
+        category: 'Icon',
+        disable: true,
+      },
+      control: {
+        type: 'select',
+        options: [
+          'rotate-90',
+          'rotate-180',
+          'rotate-270',
+          'flip-horizontal',
+          'flip-vertical',
+        ],
+      },
+    },
+    icon_position: {
+      name: 'icon position',
+      type: { name: 'inline-radio', required: false },
+      defaultValue: 'after',
+      description: 'Icon position inside the button',
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: 'after' },
+        category: 'Icon',
+      },
+      control: {
+        type: 'inline-radio',
+        options: ['before', 'after'],
+      },
+    },
+    disabled: {
+      name: 'disabled',
+      type: { name: 'boolean', required: false },
+      defaultValue: false,
+      description: 'Disabled button',
+      table: {
+        type: { summary: 'boolean' },
+        defaultValue: { summary: false },
+        category: 'States',
+      },
+      control: {
+        type: 'boolean',
+      },
+    },
+  },
+  decorators: [withCode, withNotes],
 };
 
-export const Primary = () => {
-  const data = prepareButton(dataPrimary);
-  const name = select('icon.name', iconsList, '', tabLabels.optional);
-  if (name !== '') {
-    getIconKnobs(data, name, 'ui', 'xs');
-  } else if (name === '' && data.icon) {
-    delete data.icon.name;
-  }
+export const Primary = (args) =>
+  button(prepareControls(Primary, dataPrimary, args));
 
-  return button(data);
+Primary.args = {
```

Comment:
If I understand correctly, `{Component}.args = ...` is a way to set defaults and let storybook infer type. So in the basic case here and in the documentation it will be string which is fine.

However, all the other properties, values and types are in `prepareControls()` and `argTypes`, so I suggest we try to move the setting of specification-based values (defaults) to one of these places too, for consistency.

Reply:
Could you please share the resolution to this one?
I mean, why isn't this part of `getArgTypes()` instead?

Reply:
I'm still wondering whether https://github.com/ec-europa/europa-component-library/blob/f196eae5e77af0150f54b19cc666f67093515dd1/src/implementations/twig/components/accordion/accordion.story.js#L15 would be possible as an approach

Reply:
Anyways, reading the docs for the controls they have the same approach, so it should be fine after all having per-variant args overriding the default's argTypes